homeassistant/components/water_heater/__init__.py

```diff
@@ -332,7 +333,7 @@ def min_temp(self):
         )
 
     @property
-    def max_temp(self):
+    def max_temp(self) -> float:
```

Comment:
Maybe this should allow `None`?
```suggestion
    def max_temp(self) -> float | None:
```